### PR TITLE
Icons: Generate type definitions

### PR DESCRIPTION
**Pull request by eliasbraun**
I'm trying to automatically generate typed components, so we don't need to type every single icon in the other packages.

Run `yarn run build` to see them being created.

### CODE REVIEW

**Comment by rejas** [comment]
I only see tsx files but no type definitions :-(

**Review by peterhass — COMMENTED** [review]
try this PR in some project that doesn't support typescript. pretty sure it will fail

**Comment by eliasbraun** [comment]
Damn, will look into it again!

File: packages/icons/scripts/build.js [hunk 1]
```diff
@@ -5,30 +5,43 @@ const chalk = require('chalk');
...
     const { data: optimizedData } = await optimize(data, { path: svgPath, ...svgoConfig });
-    const reactComponentCode = await transform(data, {}, {
-      componentName: getComponentName(fileName),
-      filePath: svgPath,
-    });
+
+    const reactComponentCode = await generateComponentCode(data, fileName, svgPath);
+
+    const withTs = true;
+    const reactTsComponentCode = await generateComponentCode(data, fileName, svgPath, withTs);
```
Comment (peterhass) [inline packages/icons/scripts/build.js:44]:
```suggestion
    const reactTsComponentCode = await generateComponentCode(data, fileName, svgPath, true);
```

or:

```suggestion
    const reactTsComponentCode = await generateComponentCode({ data, fileName, svgPath, withTypescript: true });
```

Please don't use abbreviations

**Review by peterhass — APPROVED** [review]

**Review by peterhass — APPROVED** [review]

**Pull request merged**